dptb/nn/energy.py

```
@@ -113,6 +123,178 @@ def forward(self,
+        self.emin = emin
+        self.emax = emax
+        self.m0 = m0
+        self.max_refinement = max_refinement
+        self.uplo = uplo
+        self.extract_triangular = extract_triangular
+        
+        # Initialize solver to check availability
+        try:
+            self.solver = FeastSolver()
+        except ImportError as e:
+            raise ImportError(f"FEAST solver not available: {e}")
```

Ruff (0.15.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)